### ISSUE TITLE
Fix remaining `using nuing = ` aliases

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#44117]
There are still a few places where we're explicitly defining nint/nuint as aliases for int/long/uint/ulong rather than using the new C# 9 types:
```C#
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Latin1Utility.cs(13):using nint = System.Int64;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Latin1Utility.cs(14):using nuint = System.UInt64;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Latin1Utility.cs(16):using nint = System.Int32;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Latin1Utility.cs(17):using nuint = System.UInt32;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Unicode\Utf16Utility.Validation.cs(18):using nuint_t = System.UInt64;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Unicode\Utf16Utility.Validation.cs(20):using nuint_t = System.UInt32;
d:\repos\runtime\src\libraries\System.Private.CoreLib\src\System\Text\Unicode\Utf16Utility.Validation.cs(23):using nuint_t = System.UInt64;
```

cc: @GrabYourPitchforks

**Comment by jkotas** [comment]
Fixing Utf16Utility.Validation.cs needs https://github.com/dotnet/runtime/issues/36160.

**Comment by GrabYourPitchforks** [comment]
`Vector<nuint_t>` is only used in one place in that file. We could use `if (IntPtr.Size == 8) { /* ... */ } else { /* ... */ }` if we really needed to. It's not that much code duplication.

**Comment by tannergooding** [comment]
This was resolved a bit back (same time as #36160 was resolved)
